### PR TITLE
Improve adjustment logic

### PR DESCRIPTION
**Pull request by rahularoradfs**
## Briefly describe the feature or bug that this PR addresses.

Improvements to test adjustment and estimate prioritization. Initially for systematic review; will be helpful on prod and for future analysis as well. 

Test adjustment changes:
- Changed the bounds for test adjustment to be somewhat more lenient in `test_adjustment.py`
- Add commented-out lines of code to `airtable_records_formatter.py` that enable test adjustment to be repeated specifically for seroprevalence estimates where it previously failed to converge, or where it wasn't considered to be bounded due to more stringent bounds in `test_adjustment.py`

Migration:
- Modify data models and create migration to remove maximum string lengths for all `varchar` dtypes in database. This is per the [messages sent this morning in #dev-backend Slack channel](https://serotracker.slack.com/archives/C019QMW4JH0/p1619274455000900), which provide a full rationale

Pooling and estimate prioritization: 
- Fix pooling code in `estimate_prioritization.py` and `pooling_functions.py` to better account for `NA` values 
- Rewrite `"analysis_dynamic"` mode option in `get_prioritized_estimates` function in `estimate_prioritization.py`

## Please link the Airtable ticket associated with this PR.

None.

## Describe the steps you took to test the feature/bugfix introduced by this PR.

- Ran migration locally
- Ran ETL locally
- Ran analysis code locally
- Ran website on localhost based on local database to test pins, filters, etc 

## Does any infrastructure work need to be done before this PR can be pushed to production?

Not sure. There is a migration involved - does that count? Otherwise, not sure what else would be required.

### CODE REVIEW

**Review by simonarocco — APPROVED** [review]
nice!!!

**Comment by rahularoradfs** [comment]
@simonarocco still needs to be completed // fully tested!! running ETL rn and playing with est prio, should be g in a couple days

**Comment by rahularoradfs** [comment]
@atatmaja @simonarocco tested and ready for your review

**Comment by rahularoradfs** [comment]
@atatmaja @simonarocco just tested with frontend locally. Everything seems to be working fine. Note that I am getting the error `'list' object has no attribute 'split'` - probably because I haven't rebased off of #203

File: app/database_etl/airtable_records_handler/airtable_records_formatter.py [hunk 1]
```diff
@@ -141,6 +142,14 @@ def add_test_adjustments(df: pd.DataFrame) -> pd.DataFrame:
     # Get all unique airtable_record_ids that are new/have been modified
     new_airtable_record_ids = diff['airtable_record_id'].unique()
 
+    # Add all unique airtable_record_ids for which test adjustment was unsuccessful
```
Comment (atatmaja) [inline app/database_etl/airtable_records_handler/airtable_records_formatter.py:145]: would we eventually want to hide this behind a flag or something

File: app/database_etl/test_adjustment_handler/test_adjustment.py [hunk 2]
```diff
@@ -24,6 +24,15 @@ def logit(p, tol = 1e-3):
     # return the logit of the constrained probability
     return log(p / (1 - p))
 
+def result_is_bounded(median_adj_prev, raw_prev, maxsmall = 0.5, minbig = 0.5):
+    logit_delta = logit(median_adj_prev) - logit(raw_prev)
+    adjusted_closeto_raw = abs(logit_delta) < 1
+    
+    both_below_maxsmall = (median_adj_prev <= maxsmall) and (raw_prev <= maxsmall)
```
Comment (atatmaja) [inline app/database_etl/test_adjustment_handler/test_adjustment.py:31]: Can you explain the significance behind these two variables here

File: app/utils/estimate_prioritization/pooling_functions.py [hunk 3]
```diff
@@ -17,6 +18,16 @@ def get_unique_value(series: pd.Series, default: Any = pd.NA) -> Any:
     else:
         return default
 
+def weighted_average(df: pd.DataFrame, values: str, weights: str) -> float:
+    df = df.dropna(axis = 'index',
```
Comment (atatmaja) [inline app/utils/estimate_prioritization/pooling_functions.py:22]: Best practice to not reassign variables that are present in the function signature.

Could call the first arg in the function signature `input_df` instead

File: app/utils/estimate_prioritization/pooling_functions.py [hunk 4]
```diff
@@ -17,6 +18,16 @@ def get_unique_value(series: pd.Series, default: Any = pd.NA) -> Any:
     else:
         return default
 
+def weighted_average(df: pd.DataFrame, values: str, weights: str) -> float:
```
Comment (atatmaja) [inline app/utils/estimate_prioritization/pooling_functions.py:21]: nit: probably clearer to name these `value_col` and `weight_col` instead

File: app/database_etl/airtable_records_handler/airtable_records_formatter.py (hunk 1 above)
Comment (rahularoradfs) [reply app/database_etl/airtable_records_handler/airtable_records_formatter.py:145]: Possible -- maybe behind a switch that we can throw when we call the ETL pipeline? 

For now, I suspect it's mostly for local use, and just valuable to have here commented out for folks to use locally as needed.

File: app/database_etl/test_adjustment_handler/test_adjustment.py (hunk 2 above)
Comment (rahularoradfs) [reply app/database_etl/test_adjustment_handler/test_adjustment.py:31]: Removed `maxsmall` and `minbig` -- these are better for fixed values. Added comments to explain.

File: app/utils/estimate_prioritization/pooling_functions.py (hunk 3 above)
Comment (rahularoradfs) [reply app/utils/estimate_prioritization/pooling_functions.py:22]: done

File: app/utils/estimate_prioritization/pooling_functions.py (hunk 4 above)
Comment (rahularoradfs) [reply app/utils/estimate_prioritization/pooling_functions.py:21]: done

**Pull request merged**